### DISCUSSION TITLE
Lubelogger but for pets & livestock

### DISCUSSION THREAD

**Discussion by DiscoSlug** [hargata/lubelog#765] [Ideas]
Hi,

This is just such a brilliant piece of software. I've been using it since a few months after it came out and watched the breakneck speed at which it was developed and new features were added. 

In addition to having many decrepit vehicles, I happen to have even more decrepit animals. I'm farmer adjacent (spouse of farmer) and as the firmly established techie of the house am saddled with the duties of handling the data. Don't get the wrong idea though, as only tech literate enough to lamely walk myself through tutorials with many more errors than trial and techie status in a farming household is a very low bar. 

Is there any chance someone would be interested in either spinning off, or teaching me to spin off a lubelogger pet flavored application that is more stripped down just to handle pets / livestock? All the bones are there to handle quite literally every single thing you would need as a pet owner / livestock manager.

**Comment by tigattack** [comment]
Whilst I would have no use for this, the idea piqued my interest. Below are my thoughts and the results of a quick bit of research.

Starting out with this idea would be simple enough, just a matter of doing find & replace on strings (e.g. `Garage` ⇾ `Barn`, `Vehicle` ⇾ `Animal`). Where it could get complex would be modifying/removing logic and features (e.g. I can't think of any livestock-related version of odometer tracking, so that would need to go, and while you could replace `Fuel` ⇾ `Feed`, the logic and specifics surrounding it would obviously need to be tweaked), adding features, etc.

Whether LubeLogger would be the best base for re-working into a livestock tracker is an interesting thought, but not one I could answer.

From a quick search, I found the following:
* [A write-up (including links to code repositories for server & client apps) about a pet tracking application someone wrote](https://dev.to/inesherminio/petlog-a-mongodb-based-app-where-you-can-keep-track-on-your-pets-life-and-info-23ma). The screenshots look good, but I'm not sure how far it was taken past a proof of concept. Nevertheless, it could be a good base to start with.
* https://github.com/Pet-Logger/Pet-Logger — Looks fairly basic from the README alone, but it's difficult to tell where it's at without trying it out.
* https://github.com/NellicusLabs/Pupdate — Archived & unmaintained, but possibly a good base?
* https://github.com/sheltermanager/asm3 — asm = Animal Shelter Manager
* [farmOS](https://farmos.org) — A more widely-scoped farm management, planning, and record-keeping application, including support for livestock asset management.

I'm sure more could be found. My search terms that turned up the results above were along the lines of "livestock management self hosted", "pet management self hosted", "pet logger self hosted", etc.

Unless any of the solutions above are complete enough to use out of the box, you will no doubt be looking at writing some code to flesh out an existing project for your needs or write one from scratch. If this ends up being the case, there are endless free learning resources around for getting started with various programming languages. YouTube is always a good resource, but far from the only choice of course.
